File: plugin/src/org/jetbrains/yesod/hamlet/highlight/HamletAnnotator.java
package org.jetbrains.yesod.hamlet.highlight;

/**
 * @author Leyla H
 */

import com.intellij.lang.annotation.AnnotationHolder;
import com.intellij.lang.annotation.Annotator;
import com.intellij.openapi.editor.DefaultLanguageHighlighterColors;
import com.intellij.psi.PsiElement;
import org.jetbrains.yesod.hamlet.psi.*;
import org.jetbrains.yesod.hamlet.psi.String;

public class HamletAnnotator implements Annotator {
    @Override
    public void annotate(PsiElement psiElement, AnnotationHolder annotationHolder) {
        if(psiElement instanceof Tag) {
            annotationHolder.createInfoAnnotation(psiElement, null)
                    .setTextAttributes(DefaultLanguageHighlighterColors.KEYWORD);
        }
        if(psiElement instanceof Backslash) {
            annotationHolder.createInfoAnnotation(psiElement, null)
                    .setTextAttributes(DefaultLanguageHighlighterColors.KEYWORD);
        }
        if(psiElement instanceof Doctype) {
            annotationHolder.createInfoAnnotation(psiElement, null).setTextAttributes(HamletColors.OPERATOR);
        }

        if(psiElement instanceof Operator) {
            annotationHolder.createInfoAnnotation(psiElement, null).setTextAttributes(HamletColors.OPERATOR);
        }
        if(psiElement instanceof Comment) {
            annotationHolder.createInfoAnnotation(psiElement, null).setTextAttributes(HamletColors.COMMENT);
        }
        if(psiElement instanceof Attribute) {
            annotationHolder.createInfoAnnotation(psiElement, null).setTextAttributes(HamletColors.ATTRIBUTE);
        }
        if(psiElement instanceof AttributeValue) {
            annotationHolder.createInfoAnnotation(psiElement, null).setTextAttributes(HamletColors.ATTRIBUTE_VALUE);
        }
        if(psiElement instanceof Interpolation) {
            annotationHolder.createInfoAnnotation(psiElement, null).setTextAttributes(HamletColors.ATTRIBUTE_VALUE);
        }
        if(psiElement instanceof EndInterpolation) {
            annotationHolder.createInfoAnnotation(psiElement, null).setTextAttributes(HamletColors.ATTRIBUTE_VALUE);
        }
        if(psiElement instanceof Dollar) {
            annotationHolder.createInfoAnnotation(psiElement, null).setTextAttributes(HamletColors.ATTRIBUTE_VALUE);
        }
        if(psiElement instanceof String) {
            annotationHolder.createInfoAnnotation(psiElement, null).setTextAttributes(HamletColors.STRING);
        }
        if(psiElement instanceof ColonIdentifier) {
            annotationHolder.createInfoAnnotation(psiElement, null).setTextAttributes(HamletColors.IDENTIFIER);
        }
        if(psiElement instanceof DotIdentifier) {
            annotationHolder.createInfoAnnotation(psiElement, null).setTextAttributes(HamletColors.IDENTIFIER);
        }
        if(psiElement instanceof SharpIdentifier) {
            annotationHolder.createInfoAnnotation(psiElement, null).setTextAttributes(HamletColors.IDENTIFIER);
        }
    }
}

File: plugin/src/org/jetbrains/yesod/hamlet/highlight/HamletAnnotator.kt
package org.jetbrains.yesod.hamlet.highlight

/**
 * @author Leyla H
 */

import com.intellij.lang.annotation.AnnotationHolder
import com.intellij.lang.annotation.Annotator
import com.intellij.openapi.editor.DefaultLanguageHighlighterColors
import com.intellij.psi.PsiElement
import org.jetbrains.yesod.hamlet.psi.*
import org.jetbrains.yesod.hamlet.psi.String

class HamletAnnotator : Annotator {
    override fun annotate(psiElement: PsiElement, annotationHolder: AnnotationHolder) {
        if (psiElement is Tag) {
            annotationHolder.createInfoAnnotation(psiElement, null).textAttributes = DefaultLanguageHighlighterColors.KEYWORD
        }
        if (psiElement is Escape) {
            annotationHolder.createInfoAnnotation(psiElement, null).textAttributes = DefaultLanguageHighlighterColors.KEYWORD
        }
        if (psiElement is Doctype) {
            annotationHolder.createInfoAnnotation(psiElement, null).textAttributes = HamletColors.OPERATOR
        }
        if (psiElement is Operator) {
            annotationHolder.createInfoAnnotation(psiElement, null).textAttributes = HamletColors.OPERATOR
        }
        if (psiElement is Comment) {
            annotationHolder.createInfoAnnotation(psiElement, null).textAttributes = HamletColors.COMMENT
        }
        if (psiElement is Attribute) {
            annotationHolder.createInfoAnnotation(psiElement, null).textAttributes = HamletColors.ATTRIBUTE
        }
        if (psiElement is AttributeValue) {
            annotationHolder.createInfoAnnotation(psiElement, null).textAttributes = HamletColors.STRING
        }
        if (psiElement is Interpolation) {
            annotationHolder.createInfoAnnotation(psiElement, null).textAttributes = HamletColors.ATTRIBUTE_VALUE
        }
        if (psiElement is String) {
            annotationHolder.createInfoAnnotation(psiElement, null).textAttributes = HamletColors.STRING
        }
        if (psiElement is ColonIdentifier) {
            annotationHolder.createInfoAnnotation(psiElement, null).textAttributes = HamletColors.IDENTIFIER
        }
        if (psiElement is DotIdentifier) {
            annotationHolder.createInfoAnnotation(psiElement, null).textAttributes = HamletColors.ATTRIBUTE
        }
        if (psiElement is SharpIdentifier) {
            annotationHolder.createInfoAnnotation(psiElement, null).textAttributes = HamletColors.ATTRIBUTE
        }
    }
}


